src/VisualStudio/Core/Def/Implementation/Diagnostics/MiscellaneousDiagnosticAnalyzerService.cs

```diff
@@ -82,7 +90,9 @@ public SyntaxOnlyDiagnosticAnalyzer(MiscellaneousDiagnosticAnalyzerService servi
                 Contract.Requires(document.Project.Solution.Workspace == _workspace);
 
                 var diagnosticData = diagnostics == null ? ImmutableArray<DiagnosticData>.Empty : diagnostics.Select(d => DiagnosticData.Create(document, d)).ToImmutableArrayOrEmpty();
-                _service.RaiseDiagnosticsUpdated(new DiagnosticsUpdatedArgs(new MiscUpdateArgsId(document.Id), _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
+                _service.RaiseDiagnosticsUpdated(
+                    new DiagnosticsUpdatedArgs(new MiscUpdateArgsId(document.Id, _service.analyzerService.GetCompilerDiagnosticAnalyzer(document.Project.Language)),
+                    _workspace, document.Project.Solution, document.Project.Id, document.Id, diagnosticData));
```

Comment:
Agreed. Would we want to distinguish between live and build here or should it simply be "VB Compiler"